clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -6036,6 +6036,57 @@ static bool canPassInRegisters(Sema &S, CXXRecordDecl *D,
   return HasNonDeletedCopyOrMove;
 }
 
+/// Add [[gsl::Owner]] and [[gsl::Pointer]] attributes for std:: types.
+/// \pre Record->getIdentifiers() is non-null
+void Sema::addOwnerPointerAttribute(CXXRecordDecl *Record) {
+
+  static llvm::StringSet<> StdOwners{
+      "any",
+      "array",
+      "deque",
+      "forward_list",
+      "vector",
+      "list",
+      "map",
+      "multiset",
+      "multimap",
+      "optional",
+      "priority_queue",
+      "queue",
+      "regex",
+      "set",
+      "stack",
+      "unordered_set",
+      "unordered_map",
+      "unordered_multiset",
+      "unordered_multimap",
+      "variant",
```

Comment:
What would be the deref type for variant? I think we should not add it now (since we do not have the gsl::lifetime annotation yet).

Reply:
As we discussed: Without doing automatic function model inference, we don't need the deref type.
Later, we add [[gsl::lifetime]] to the `std::get()` function to make that work.
The lifetime paper explicitly says 
```
Additionally, for convenient adoption without modifying existing standard library headers, the following well-known standard types are treated as-if annotated as Owners: stack, queue, priority_queue, optional, variant, any, and regex.
```